forest/intake_loader.py

```diff
@@ -0,0 +1,395 @@
+                                                     selected_time),
+                           'latitude': lat_filter,
+                           }
+        coord_names = [c1.name() for c1 in cube.coords()]
+        if 'air_pressure' in coord_names:
+            constraint_dict['air_pressure'] = functools.partial(
+                pressure_select,
+                pressure,
+            )
+        cube_cropped = cube.extract(iris.Constraint(**constraint_dict))
+        lat_pts = cube_cropped.coord('latitude').points
+        long_pts = cube_cropped.coord('longitude').points - 180.0
+        cube_data_cropped = cube_cropped.data
+        cube_width = int(cube_data_cropped.shape[1] / 2)
+        cube_data_cropped = numpy.concatenate(
+            [cube_data_cropped[:, cube_width:],
+             cube_data_cropped[:, :cube_width]], axis=1)
+
+        data = geo.stretch_image(long_pts, lat_pts, cube_data_cropped)
+        data['image'] = [numpy.ma.masked_array(data['image'][0],
+                                               mask=numpy.isnan(
+                                                   data['image'][0]))]
+        return data
+
+class IntakeLoader:
+    """
+    Loader class for the CMIP6 intake dataset.
+    """
+    def __init__(self, pattern):
+        institution_id, experiment_id,member_id, grid, table_id,activity_id = pattern.split('_')
+        self.experiment_id = experiment_id
+        self.table_id = table_id
+        self.grid_label = grid
+        self.variable_id = ''
+        self.institution_id = institution_id
+        self.activity_id = activity_id
+        self.member_id = member_id
+        self._label = f'{self.experiment_id}_{self.institution_id}_{self.member_id}'
+        self._cube = None
+
+    @property
+    def cube(self):
+        """
+        The purpose of this property is to delay loading of the cube until the
+        point where all relevant parameters are defined and data and metadata
+        can be downloaded.
+        """
+        if not self._cube:
+            self._load_cube()
+        return self._cube
+
+    def _load_cube(self):
+        self._cube = _load_from_intake(experiment_id=self.experiment_id,
+                                       table_id=self.table_id,
+                                       grid_label=self.grid_label,
+                                       variable_id=self.variable_id,
+                                       institution_id=self.institution_id,
+                                       activity_id=self.activity_id,
+                                       member_id=self.member_id)
+
+    def image(self, state):
+        """
+        Main image loading function. This function will actually realise the
+        data,
+        """
+        do_update = False
+        if self.variable_id != state.variable:
+            self.variable_id = state.variable
+            do_update = True
+
+        if do_update:
+            self._cube = None
```

Comment:
This can be simplified to a single `if` block without the need for `do_update`:
```python
if self.variable_id != state.variable:
    self.variable_id = state.variable:
    self._cube = None
```

Reply:
Refactored as suggested in c1403bf